explorer-backend/transactions/evm.go

```diff
@@ -99,10 +99,50 @@ func (w *EVMWatcher) fetchEventsFromBlockRange(ctx context.Context, fromHeight u
 		return nil
 	}
 
-	for height := fromHeight; height < toHeight; height++ {
-		if err := w.fetchEventsByBlockNumber(ctx, height); err != nil {
+	client := w.connector.Client()
+	query := eth.FilterQuery{
+		FromBlock: big.NewInt(int64(fromHeight)),
+		ToBlock:   big.NewInt(int64(toHeight)),
+		Addresses: []ethCommon.Address{*w.contractAddress},
+		Topics:    [][]ethCommon.Hash{{ethereum.LogMessagePublishedTopic}},
+	}
```

Comment:
Since there are many blocks at the first startup, I just tested it on the testnet, if it is very slow to get one by one, so I changed it to get events in batches.

Reply:
1. No, for the EVM chains, we query from the height of contract deployments; for Alephium, we query from the event index 0.
2. In order to avoid losing events, it will query every reboot, but not from the begining. For example, if we deploy the contracts at a block with a height of `A`, then the explorer backend stops running at a height of `B`, and restarts at a height of `C`, where `C > B > A`, then it will start querying events from B instead of start with A

Reply:
If the explorer backend stops at a height of `B` with a block `b1` and there are two blocks `b0` and `b1` at the same height with `b0` becomes the cannonical block eventually, does the sync function work well after rebooting?

Reply:
No, but I think Alephium does not have this problem because we query from the last event index.

For EVM chains, I think we can start the query event from the height `B - N` when restarting, where N can be 64(2 epochs) for ETH. WDYT?

Reply:
Yes, that sounds great to me!